pulp_ansible/app/utils.py

```diff
@@ -0,0 +1,26 @@
+from django.db.models import Q
+from pulpcore.plugin.models import RepositoryVersion, RepositoryContent
+
+
+def filter_content_for_repo_version(qs, repo_version):
+    """
+    Returns the given queryset, but filtered to only include content
+    from the specified repository version.
+
+    qs: Content queryset. Must be a model that is or inherits from
+        pulpcore.plugin.models.Content
+    repo_version: repository version to return content from
+
+    This generally seems to be faster than repo_version.get_content()
+    """
+
+    repo_version_qs = RepositoryVersion.objects.filter(
+        repository=repo_version.repository, number__lte=repo_version.number
+    ).values_list("pk")
+
+    f = Q(version_added__in=repo_version_qs) & Q(
+        Q(version_removed=None) | ~Q(version_removed__in=repo_version_qs)
+    )
+    content_rel = RepositoryContent.objects.filter(f)
```

Comment:
In my testing this was slightly more efficient. The key here is `Q(version_removed=None) | ~Q(version_removed__in=repo_version_qs)`. If the version remove is none, then it prevents the lookup in the `repo_version_qs` in postgres from happening. When you use `.exclude()` it will always perform the `version_removed__in` lookup

Reply:
I think you should add `Q(repository=repo_version.repository_id)`  to the filter.